src/handlers/inviteLink.ts

```diff
@@ -0,0 +1,126 @@
+                        hasRole = true;
+                    }
+                });
+            } catch (e) {
+                Logger.error(e);
+            }
+
+            if (hasRole) {
+                const allowedEmbed = makeEmbed({
+                    title: 'Invite Link Detected',
+                    thumbnail: { url: 'https://cdn.discordapp.com/attachments/932350968522240101/932625886275043338/Approved.png' },
+                    description: 'An allowed role has sent an invite link',
+                    author: {
+                        name: msg.author.tag,
+                        iconURL: msg.author.displayAvatarURL({ dynamic: true }),
+                    },
+                    fields: [
+                        {
+                            name: 'User:',
+                            value: `${msg.author}`,
+                        },
+                        {
+                            name: 'Channel:',
+                            value: `${msg.channel}`,
+                        },
+                        {
+                            name: 'Message Content:',
+                            value: codeBlock(msg.content.toString()),
+                        },
+                    ],
+                });
+
+                await modLogs.send({ embeds: [allowedEmbed] });
+                return;
+            }
+
+            await msg.delete();
+
+            const notAllowedEmbed = makeEmbed({
+                title: 'Invite Link Detected',
+                thumbnail: { url: 'https://cdn.discordapp.com/attachments/932350968522240101/932625893657026630/Scam.png' },
+                author: {
+                    name: msg.author.tag,
+                    iconURL: msg.author.displayAvatarURL({ dynamic: true }),
+                },
+                fields: [
+                    {
+                        name: 'User:',
+                        value: `${msg.author}`,
+                    },
+                    {
+                        name: 'Channel:',
+                        value: `${msg.channel}`,
+                    },
+                    {
+                        name: 'Message Content:',
+                        value: codeBlock(msg.content.toString()),
+                    },
+                ],
+            });
+            try {
+                await msg.member.timeout(60 * 60 * 24 * 1000, 'Invite Link');
+            } catch (e) {
+                Logger.error(e);
+                const errorEmbed = makeEmbed({
+                    title: 'Error timing out user',
+                    description: makeLines([
+                        `An error occurred while timing out ${msg.author}`,
+                        `${codeBlock(`Error : ${e}`)}`,
+                    ]),
+                    color: Colors.Red,
+                });
+                await modLogs.send({ embeds: [errorEmbed] });
+
+                await modLogs.send({ embeds: [notAllowedEmbed] });
```

Comment:
Could put sending the notAllowedEmbed before trying to timeout and before trying to DM. That way you don't have to call it in every catch block?